bootstrap/config/config.go

```diff
@@ -625,47 +627,54 @@ func (cp *Processor) listenForCommonChanges(fullServiceConfig interfaces.Configu
 
 				// Config Provider sends an update as soon as the watcher is connected even though there are not
 				// any changes to the configuration. This causes an issue during start-up if there is an
-				// envVars override of one of the Writable fields, so we must ignore the first update.
+				// envVars override of one of the Writable fields, so on the first update we can just save a copy of the
+				// common writable for comparison for future writable updates.
 				if isFirstUpdate {
 					isFirstUpdate = false
+					previousCommonWritable = raw
 					continue
 				}
 
-				privateConfig, err := copyConfigurationStruct(fullServiceConfig)
-				if err != nil {
-					lc.Errorf("could not make private copy of config while watching for common config writable: %s", err.Error())
-					return
+				if err := cp.processCommonConfigChange(fullServiceConfig, previousCommonWritable, raw, privateConfigClient); err != nil {
+					lc.Error(err.Error())
 				}
+				// ensure that the local copy of the common writable gets updated no matter what
+				previousCommonWritable = raw
+			}
+		}
+	}()
+}
 
-				err = cp.loadConfigFromProvider(privateConfig, privateConfigClient)
-				if err != nil {
-					lc.Errorf("could not load private config while watching for common config writable: %s", err.Error())
-					return
-				}
+func (cp *Processor) processCommonConfigChange(fullServiceConfig interfaces.Configuration, previousCommonWritable any, raw any, privateConfigClient configuration.Client) error {
+	privateConfig, err := copyConfigurationStruct(fullServiceConfig)
+	if err != nil {
+		return fmt.Errorf("could not make private copy of config while watching for common config writable: %s", err.Error())
+	}
 
-				// check if changed value is a private override
-				if cp.isPrivateOverride(fullServiceConfig.GetWritablePtr(), raw, privateConfigClient) {
-					continue
-				}
+	err = cp.loadConfigFromProvider(privateConfig, privateConfigClient)
+	if err != nil {
+		return fmt.Errorf("could not load private config while watching for common config writable: %s", err.Error())
+	}
```

Comment:
Not sure we need this. fullServiceConfig should be the latest "previous" config..

Reply:
this is not needed because we pull the private config keys from the config provider